src/main/java/com/rackspacecloud/metrics/queryservice/controllers/QueryController.java

```diff
@@ -28,263 +21,16 @@
-        LOGGER.info("tags: request received for tenantId [{}], measurement [{}]", tenantId, measurement);
-
-        List<QueryDomainOutput> out = queryService.tags(tenantId, measurement);
-        List<String> response = new ArrayList<>();
-        out.forEach(queryResult -> response.addAll(queryResult.getValuesAsCollection()));
-
-        return response;
-    }
-
-    /**
-     * Get all of the values for a given tag in a measurement
-     * @param tenantId tenantId who has access to this endpoint
-     * @param measurement measurement name
-     * @param tagsAsCsv comma separated tag names
-     * @return
-     */
-    @RequestMapping(
-            value = "/{tenantId}/measurements/{measurement}/tags/{tagsAsCsv}/values",
-            method = RequestMethod.GET,
-            produces = MediaType.APPLICATION_JSON_VALUE
-    )
-    @Secured({"ROLE_COMPUTE_DEFAULT"})
-    @Timed(value = "query.service", extraTags = {"query.type","tags.values"})
-    public List<?> tagValues(
-            @NotBlank @PathVariable final String tenantId,
-            @NotBlank @PathVariable final String measurement,
-            @NotBlank @PathVariable final String tagsAsCsv) {
-        LOGGER.info("tagValues: request received for tenantId [{}], measurement [{}] and tag [{}]",
-                tenantId, measurement, tagsAsCsv);
-
-        List<QueryDomainOutput> out = queryService.tagValues(tenantId, measurement, tagsAsCsv);
-        List<Map<String, Object>> response = new ArrayList<>();
-        out.forEach(queryResult -> response.addAll(queryResult.getQueryResponse()));
-
-        return response;
-    }
-
-    /**
-     * Get all of the fields for given measurement
-     * @param tenantId tenantId who has access to this endpoint
-     * @param measurement measurement name
-     * @return
-     */
-    @RequestMapping(
-            value = "/{tenantId}/measurements/{measurement}/fields",
-            method = RequestMethod.GET,
-            produces = MediaType.APPLICATION_JSON_VALUE
-    )
-    @Secured({"ROLE_COMPUTE_DEFAULT"})
-    @Timed(value = "query.service", extraTags = {"query.type","fields"})
-    public List<?> fields(
-            @NotBlank @PathVariable final String tenantId,
-            @NotBlank @PathVariable final String measurement) {
-        LOGGER.info("fields: request received for tenantId [{}], measurement [{}]", tenantId, measurement);
-
-        List<QueryDomainOutput> out = queryService.fields(tenantId, measurement);
-        List<Map<String, Object>> response = new ArrayList<>();
-        out.forEach(queryResult -> response.addAll(queryResult.getQueryResponse()));
-
-        return response;
-    }
-
-    private QueryDomainInput inputToDomainInput(QueryInput queryInput) {
-        QueryDomainInput domainInput = new QueryDomainInput();
-        domainInput.setQueryString(queryInput.getQueryString());
-        return domainInput;
-    }
-
     @RequestMapping(
             value = "/query",
             method = RequestMethod.GET,
```

Comment:
As a convenience, Spring Boot provides `@GetMapping`. You also don't need the "produces JSON" since `@RestController` on class defaults to that.